addons/event/report/event_event_templates.xml

```diff
@@ -60,7 +60,11 @@
                                             <span t-out="12345678901234567890" class="barcode ms-2" t-options="{'widget': 'barcode', 'width': 200, 'height': 84, 'quiet': 0, 'humanreadable': 1}"/>
                                         </t>
                                     </t>
-                                    <span><img class="ms-2" t-attf-src="/report/barcode/QR/{{attendee.barcode if attendee else '12345678901234567890'}}?&amp;width=116&amp;height=116&amp;quiet=0" alt="QR Code"/></span>
+                                    <span>
+                                        <t t-if="not attendee or (attendee and object.barcode )">
+                                            <img class="ms-2" t-attf-src="/report/barcode/QR/{{object.barcode if object.barcode else '12345678901234567890'}}?&amp;width=116&amp;height=116&amp;quiet=0" alt="QR Code"/>
+                                        </t>
```

Comment:
This fails because this is a report on event.event, so object doesn't have a barcode. 'attendee' does